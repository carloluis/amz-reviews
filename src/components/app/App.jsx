import React from 'react';
import Reviews from '../reviews';
import SearchFilter from '../search';
import StarsFilter from '../stars';
import styles from './App.scss';

// TODO: filters

const App = () => (
    <div className={styles.container}>
        <header>
            <h1>Reviews Demo</h1>
        </header>
        <section className={styles.filters}>
            <SearchFilter />
            <StarsFilter />
        </section>
        <Reviews />
    </div>
);

export default App;
