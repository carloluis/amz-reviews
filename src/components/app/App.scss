@import '../../styles/palette.scss';

.container {
    max-width: 95%;
    margin: 16px auto;
    padding: 8px;

    & > header {
        color: $dark-font;
    }

    .filters {
        position: sticky;
        top: -2px;
        background: $white;
        padding: 16px 0;
        opacity: 0.95;
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        height: 144px;
    }
}

@media screen and (min-width: 769px) {
    .container {
        max-width: 760px;

        .filters {
            padding-right: 300px;
        }
    }
}
