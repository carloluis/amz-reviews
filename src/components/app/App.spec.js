import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import App from './App';

describe('<App/>', () => {
    it('snapshot', () => {
        const renderer = new ShallowRenderer();
        renderer.render(<App />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
});
