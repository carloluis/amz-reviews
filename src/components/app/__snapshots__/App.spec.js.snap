// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> snapshot 1`] = `
<div
  className="container"
>
  <header>
    <h1>
      Reviews Demo
       
      <a
        href="https://github.com/carloluis/amz-reviews"
        rel="noopener noreferrer"
        target="_blank"
        title="GitHub Project"
      >
        ðŸ“ƒ
      </a>
    </h1>
  </header>
  <section
    className="filters"
  >
    <ErrorBoundary>
      <Connect(Search) />
      <div
        className="grouping"
      >
        <Connect(GroupBy) />
        <Connect(OrderBy) />
      </div>
      <Connect(Stars) />
    </ErrorBoundary>
  </section>
  <ErrorBoundary>
    <Connect(Reviews) />
  </ErrorBoundary>
</div>
`;
